### PR TITLE
property now outputs empty xml property strings without offset

### PR DESCRIPTION
**Pull request by JensWehner**
before: `<value attr1="bla">              </value>`
now:    `<value attr1="bla"/>`

### CODE REVIEW

**Comment by ruehle** [comment]
why not
<value attr1="bla"/>

File: CMakeLists.txt [hunk 1]
```diff
@@ -60,7 +60,7 @@ endif(IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/.git)
 find_package(Threads REQUIRED)
 set(THREAD_LIBRARIES ${CMAKE_THREAD_LIBS_INIT})
 
-find_package(Boost 1.39.0 REQUIRED COMPONENTS program_options filesystem system )
+find_package(Boost 1.57.0 REQUIRED COMPONENTS program_options filesystem system )
```
Comment (JoshuaSBrown) [inline CMakeLists.txt]: Does this mean that Boost 1.57.0 is the minimum required version?

File: src/libtools/property.cc [hunk 2]
```diff
@@ -179,88 +177,87 @@ void PrintNodeTXT(std::ostream &out, Property &p, const int start_level, int lev
...
+      string cAttributeValue = color->Green();
+      string cReset = color->Reset();
+
+      // print starting only from the start_level (the first node (level 0) can be <> </>)
+      if (level >= start_level) {
+        // print the node name
+        out << indent << offset << "<" << cKey << p.name() << cReset;
+        // print the node attributes 
+        for (ia = p.firstAttribute(); ia != p.lastAttribute(); ++ia) {
+          out << " "
```
Comment (JoshuaSBrown) [inline src/libtools/property.cc]: Can you clean up the indents?

File: src/libtools/property.cc [hunk 3]
```diff
@@ -179,88 +177,87 @@ void PrintNodeTXT(std::ostream &out, Property &p, const int start_level, int lev
...
+      }
+
+      // continue iteratively through the rest of the nodes
+      for (Property& prop : p) {
+        level++;
+        if (level > start_level) offset += "\t";
+        PrintNodeXML(out, prop, piom, level, offset);
+        if (level > start_level) offset.resize(offset.size() - 1);
+        level--;
+      }
```
Comment (JoshuaSBrown) [inline src/libtools/property.cc]: This doesn't really matter but just for your info: I believe preincremeinting is supposed to be faster than post though some people argue it doesn't really matter. 
https://softwareengineering.stackexchange.com/questions/358725/pre-increment-vs-post-increment

```
--level;
++level;
```

File: netbeans/libtools/nbproject/configurations.xml [hunk 4]
```diff
@@ -40,7 +39,6 @@
       <itemPath>../../include/votca/tools/vec.h</itemPath>
       <itemPath>../../include/votca/tools/version.h</itemPath>
       <itemPath>../../include/votca/tools/votca_config.h.in</itemPath>
-      <itemPath>../../include/votca/tools/votca_gsl_boost_ublas_matrix_prod.h</itemPath>
```
Comment (JoshuaSBrown) [inline netbeans/libtools/nbproject/configurations.xml]: We are no longer using ublas because we switched to eigen right?

Comment (JensWehner) [reply netbeans/libtools/nbproject/configurations.xml]: yes

File: src/libtools/property.cc (hunk 3 above)
Comment (JensWehner) [reply src/libtools/property.cc]: for simple data types it makes no difference

File: CMakeLists.txt (hunk 1 above)
Comment (JensWehner) [reply CMakeLists.txt]: yes. the rest of votca has that already

**Pull request merged**